presto-main/src/main/java/com/facebook/presto/operator/ArrayPositionLinks.java

```diff
@@ -47,9 +52,25 @@ public int link(int left, int right)
         }
 
         @Override
-        public Factory build()
+        public PositionLinks.Factory build()
         {
-            return filterFunction -> new ArrayPositionLinks(positionLinks);
+            return new PositionLinks.Factory()
+            {
+                @Override
+                public PositionLinks create(Optional<JoinFilterFunction> joinFilterFunction)
+                {
+                    return new ArrayPositionLinks(positionLinks);
+                }
+
+                @Override
+                public HashCode checksum()
```

Comment:
We always have hash/checksum return a long (I don't think we have ever used the Guava stuff), so stick with that.